modules/swagger-codegen/src/main/resources/go/api.mustache

```diff
@@ -42,15 +41,27 @@ func (a {{classname}}) {{nickname}} ({{#allParams}}{{paramName}} {{{dataType}}}{
   var httpMethod = "{{httpMethod}}"
  // create path and map variables
   path := a.Configuration.BasePath + "{{path}}"
-{{#pathParams}} path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
+{{#pathParams}}  path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
 {{/pathParams}}
 
   {{#allParams}}
   {{#required}}
   // verify the required parameter '{{paramName}}' is set
-  if &{{paramName}} == nil {
-      return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
+  {{#isListContainer}}
+  if len({{paramName}}) == 0 {
+       return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), fmt.Errorf("missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
```

Comment:
I suggest we use `errors.New("missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")` instead


Ref: http://spf13.com/presentation/7-biggest-mistakes-in-go/, starting at slide 19